Client/mods/deathmatch/logic/CClientColModel.h

```diff
@@ -22,17 +24,68 @@ class CClientColModel : public CClientEntity
 
     eClientEntityType GetType() const { return CCLIENTCOL; }
 
-    bool LoadCol(const SString& strFile, bool bIsRawData);
-    bool Load(bool isRaw, SString input);
-
-    bool IsLoaded() { return m_pColModel != NULL; };
+    bool           LoadCol(const SString& strFile, bool bIsRawData);
+    bool           Load(bool isRaw, SString input);
+    unsigned short GetVerticesCount() const { return m_usVerticesCount; };
+    void           SetCollisionHasChanged(bool bChanged) { b_hasChanged = bChanged; };
+    bool           HasChanged() const { return b_hasChanged; };
+    bool           IsLoaded() { return m_pColModel != NULL; };
 
     bool Replace(unsigned short usModel);
     void Restore(unsigned short usModel);
     void RestoreAll();
 
     bool        HasReplaced(unsigned short usModel);
     static bool IsCOLData(const SString& strData);
+    static bool CheckVector(CVector& vec, float fRadius = 0)
+    {
+        if (fRadius > 0)
+        {
+            return (MAX_COLLISION_SIZE >= vec.fX + fRadius && vec.fX + fRadius >= -MAX_COLLISION_SIZE && MAX_COLLISION_SIZE >= vec.fY + fRadius &&
+                    vec.fY + fRadius >= -MAX_COLLISION_SIZE && MAX_COLLISION_SIZE >= vec.fZ + fRadius && vec.fZ + fRadius >= -MAX_COLLISION_SIZE) &&
+                   (MAX_COLLISION_SIZE >= vec.fX - fRadius && vec.fX - fRadius >= -MAX_COLLISION_SIZE && MAX_COLLISION_SIZE >= vec.fY - fRadius &&
+                    vec.fY - fRadius >= -MAX_COLLISION_SIZE && MAX_COLLISION_SIZE >= vec.fZ - fRadius && vec.fZ - fRadius >= -MAX_COLLISION_SIZE);
+        }
+        else
+        {
+            return (MAX_COLLISION_SIZE >= vec.fX && vec.fX >= -MAX_COLLISION_SIZE && MAX_COLLISION_SIZE >= vec.fY && vec.fY >= -MAX_COLLISION_SIZE &&
+                    MAX_COLLISION_SIZE >= vec.fZ && vec.fZ >= -MAX_COLLISION_SIZE);
+        }
+    }
+    static bool CheckMoveVector(CVector& vec, float fRadius = 0)
+    {
+        if (fRadius > 0)
+        {
+            return ((MAX_COLLISION_SIZE2 >= vec.fX + fRadius >= -MAX_COLLISION_SIZE2 && MAX_COLLISION_SIZE2 >= vec.fY + fRadius >= -MAX_COLLISION_SIZE2 &&
+                     MAX_COLLISION_SIZE2 >= vec.fZ + fRadius >= -MAX_COLLISION_SIZE2) &&
+                    (MAX_COLLISION_SIZE2 >= vec.fX - fRadius >= -MAX_COLLISION_SIZE2 && MAX_COLLISION_SIZE2 >= vec.fY - fRadius >= -MAX_COLLISION_SIZE2 &&
+                     MAX_COLLISION_SIZE2 >= vec.fZ - fRadius >= -MAX_COLLISION_SIZE2));
+        }
+        else
+        {
+            return (MAX_COLLISION_SIZE2 >= vec.fX && vec.fX >= -MAX_COLLISION_SIZE2 && MAX_COLLISION_SIZE2 >= vec.fY && vec.fY >= -MAX_COLLISION_SIZE2 &&
+                    MAX_COLLISION_SIZE2 >= vec.fZ && vec.fZ >= -MAX_COLLISION_SIZE2);
+        }
+    }
+    static bool CompareVector(CVector& vecMin, CVector& vecMax) { return vecMax.fX >= vecMin.fX && vecMax.fY >= vecMin.fY && vecMax.fZ >= vecMin.fZ; }
+    static void AlignVector(CVector& destMin, CVector& destMax, const CVector& src)
```

Comment:
Add comments above both these lines explaining what exactly they do
```suggestion

    // CompareVector returns true if all points in the second vector are greater than the corresponding points in the first vector
    static bool CompareVector(CVector& vecMin, CVector& vecMax) { return vecMax.fX >= vecMin.fX && vecMax.fY >= vecMin.fY && vecMax.fZ >= vecMin.fZ; }

    // AlignVector clamps a "source" vector between a min and max vector. Or something like that. I don't know.
    static void AlignVector(CVector& destMin, CVector& destMax, const CVector& src)
```